packages/devtools_app/lib/src/memory/flutter/memory_screen.dart

```diff
@@ -33,15 +36,32 @@ class MemoryBody extends StatefulWidget {
   MemoryBodyState createState() => MemoryBodyState();
 }
 
-class MemoryBodyState extends State<MemoryBody> {
-  // Creation of the controller must be in the state.
-  final MemoryController memoryController = MemoryController();
-
+class MemoryBodyState extends State<MemoryBody> with AutoDisposeMixin {
   @override
   void initState() {
+    _updateListeningState();
+
     super.initState();
   }
 
+  @override
+  void dispose() {
+    // TODO(terry): make my controller disposable via DisposableController and dispose here.
+    super.dispose();
+  }
+
+  @override
+  void didChangeDependencies() {
+    super.didChangeDependencies();
+    final MemoryController memoryController = Controllers.of(context).memory;
+
+    // Process each timeline frame.
+    addAutoDisposeListener(memoryController.memoryTimeline.sampleAddedNotifier,
+        () {
+      setState(() {});
```

Comment:
If nothing in this widget depends on the new samples on the controller, then this listener can be moved to the MemoryChart.

Reply:
Yep it's in MemoryChart.